### ISSUE TITLE
Incorrect SQL Result when using SELECT IN

### ISSUE THREAD

**Issue by vsspt** [orientechnologies/orientdb#7247]
## OrientDB Version, operating system, or hardware.
v2.2.17

## Operating System
- [x ] Windows 8

## JVM
- 1.8.0_101-b13

## Frameworks/Libs
- Spring Boot 1.5.2 

## Expected behavior and actual behavior
When executing a SELECT IN Query, the fields are mapped incorrectly.

## Steps to reproduce the problem

Please download this test project here: https://meocloud.pt/link/1086b248-ce9f-4305-abad-6ab99d01525a/2-2-17-Bug-2.zip/

In order to use the project, you will need to change the plocal url in the file: srt\test\testContext.xml.

Basically, this is a simple Spring application that executes a simple Count Query:

```
	private final String dmlSelect = "SELECT * FROM TABLE_CONFIG WHERE RETRIES < 5 AND CONFIG_KEY IN (SELECT CONFIG_KEY FROM TEST_TABLE WHERE STATUS = 0)";	

	public List<Map<String, Object>> select() {
		return getJdbcTemplate().queryForList(dmlSelect);
	}
```

The Database has the following tables:

**TABLE_CONFIG** 
_CONFIG_KEY| CONFIG_VALUE | RETRIES | FIELD_1 | FIELD_2_
1                    | AAAA                 | 2            | 1           | 2
2                    | BBBB                   | 6            | 1           | 2

**TEST_TABLE ** 
_CONFIG_KEY| STATUS
1                    |  0
2                    |  1

- When executing from the **Java Code**, the result of the Query **is mapped incorrectly**: [{CONFIG_KEY=1, **RETRIES=AAAA, CONFIG_VALUE=2**, FIELD_1=1, FIELD_2=2}]

- When executing from **OrientDB console**, the result is is **mapped correctly**: [{CONFIG_KEY=1, **RETRIES=2, CONFIG_VALUE=AAAA**, FIELD_1=1, FIELD_2=2}]

**Comment by vsspt** [comment]
It seems that the problem is not specific to the SELECT IN function, but instead the way OrientDB is mapping the SELECT *.

If replacing the "SELECT *" for "SELECT FIELD_1, FIELD_2, etc", it works as expected : 
```
private final String dmlSelect = "SELECT CONFIG_KEY, RETRIES, CONFIG_VALUE, FIELD_1, FIELD_2 FROM TABLE_CONFIG WHERE RETRIES < 5 AND CONFIG_KEY IN (SELECT CONFIG_KEY FROM TEST_TABLE WHERE STATUS = 0)";
```

**Comment by luigidellaquila** [comment]
Hi @vsspt 

I think I got the problem, probably the JDBC driver is not mapping correctly the columns to the property names. Let me assign it to @robfrank for a check

Thanks

Luigi

**Comment by robfrank** [comment]
thank you very much for your detailed report. I found the bug and I'm working on it. I've got only a red test right now, so I'm confident to be able to include the fix in the upcoming 2.2.18. I will close the issue when the test will be green .

**Comment by vsspt** [comment]
Thank you.
